### PR TITLE
Fixed website links to point to stable release docs

### PR DESCRIPTION
**Pull request by Innixma**
*Issue #, if available:*

*Description of changes:*

- Updated autogluon.mxnet.io -> auto.gluon.ai
- Updated links to point to /dev/ or /stable/ appropriately, instead of root.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by gradientsky — APPROVED** [review]

**Comment by szha** [comment]
Job PR-749-1 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-749/1/index.html

**Comment by jwmueller** [comment]
In the docs navbar, should we change "Nightly Version Documentation" -> "Dev Version Documentation"?

It is referred to as dev version everywhere else.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -43,7 +43,7 @@ GitHub provides additional document on [forking a repository](https://help.githu
 
 ## Tips for Modifying the Source Code
 
-- Using a fresh virtualenv, install the package via [these instructions](https://autogluon.mxnet.io/install.html).
+- Using a fresh virtualenv, install the package via [these instructions](https://auto.gluon.ai/dev/install.html).
 Be sure to select the *Source* option from the installation preferences, and install the package after cloning this repository via:
 ```
 python setup.py develop
```
Comment (jwmueller) [inline CONTRIBUTING.md:49]: The install instruction is now wrong? Should it instead be: `./full_install.sh`

**Review by jwmueller — APPROVED** [review]

File: CONTRIBUTING.md (hunk 1 above)
Comment (Innixma) [reply CONTRIBUTING.md:49]: Removed the install instructions here, now users should just use the link to get the website install instructions.

**Comment by Innixma** [comment]
> In the docs navbar, should we change "Nightly Version Documentation" -> "Dev Version Documentation"?
> 
> It is referred to as dev version everywhere else.

Updated to 'Dev'

**Comment by szha** [comment]
Job PR-749-2 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-749/2/index.html

**Pull request merged**